components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/messaging_reference.adoc

```diff
@@ -0,0 +1,589 @@
+
+[discrete]
+==== Parameters
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`bytes32 _msgHash_`:: The message hash.
+
+[discrete]
+==== Returns
+
+Returns `uint256 _result_`, where `_result_` is one of the following:
+[horizontal,labelwidth="30",role=stripes-odd]
+`1`:: A pending message is associated with the `_msgHash_` parameter.
+`0`:: No pending message is associated with the `_msgHash_` parameter.
+
+[discrete]
+==== Function definition
+
+Contract: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol[`StarknetMessaging.sol`]
+
+* Function: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol#L43[`l2ToL1Messages`]
+
+'''
+
+[#messageCancellationDelay]
+=== `messageCancellationDelay`
+
+[discrete]
+==== Description
+
+Returns the time interval after which you can cancel a message starting from the moment of calling the xref:#startL1ToL2MessageCancellation[`startL1ToL2MessageCancellation`] function. You can get the real value by calling the link:https://etherscan.io/address/0xc662c410c0ecf747543f5ba90660f6abebd9c8c4#readProxyContract#F11[`messageCancellationDelay`] function on etherscan.
+
+[discrete]
+==== State Mutability
+
+`view`
+
+[discrete]
+==== Parameters
+
+None.
+
+[discrete]
+==== Returns
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`uint256 _result_`:: The time interval.
+
+[discrete]
+==== Function definition
+
+Contract: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol[`StarknetMessaging.sol`]
+
+* Function: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol#L59[`messageCancellationDelay`]
+
+'''
+
+[#sendMessageToL2]
+=== `sendMessageToL2`
+
+[discrete]
+==== Description
+
+Sends a message to an L2 contract. The message fee is the cost of executing this function.
+
+[discrete]
+==== State Mutability
+
+`payable`
+
+[discrete]
+==== Parameters
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`uint256 _toAddress_`:: The address of the L2 contract.
+`uint256 _selector_`::
+include::messaging_reference.adoc[tag=uint256_selector]
+`uint256[] calldata _payload_`:: The payload of the message.
+
+[discrete]
+==== Returns
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`bytes32 _msgHash_`:: The hash of the message.
+`uint256 _nonce_`:: The nonce of the message.
+
+[discrete]
+==== Function definition
+
+Contract: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol[`StarknetMessaging.sol`]
+
+* Function: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol#L110[`sendMessageToL2`]
+
+'''
+
+[#startL1ToL2MessageCancellation]
+=== `startL1ToL2MessageCancellation`
+
+[discrete]
+==== Description
+
+Starts the cancellation of an L1 to L2 message.
```

Comment:
@xiaolou86 Be consistent with how you refer to these messages. See next comment for more info.

Reply:
```suggestion
Starts the cancellation of a message from L1 to L2.
```

Reply:
Done.